### PR TITLE
[s] Golden Bike Horn Fix

### PR DESCRIPTION
**Pull request by Mercenaryblue**
Will flip mobs before the return

### CODE REVIEW

**Comment by optimumtact** [comment]
nice merge bro :smile:

**Comment by coiax** [comment]
https://git-scm.com/book/en/v2

**Comment by phil235** [comment]
any reason why we need this?

**Comment by Mercenaryblue** [comment]
The lack of return ..() on the bike horn's main attack means with current code, it will hit return before the flip_mobs().

**Comment by phil235** [comment]
> it will hit return before the flip_mobs()

But there are no return in that proc... fliph_mobs() will get called whether ..() is called after or before it.

**Comment by coiax** [comment]
phil's right, why are you making this change? Have you spotted a bug in game or something?

**Comment by optimumtact** [comment]
also technically if you do call parent you should do 

. = ..()

so that the value is returned if you don't explicitly override what the parent proc returns.

**Comment by Mercenaryblue** [comment]
I think I did, yes. The flip_mobs wouldn't proc on the use // attack_self() proc.

My theory was the code would run the attack_self() proc first, and hit the return on 125 before going for flip_mobs(), thus preventing the flip_mobs() code from be read.

My fix was to either make the flip_mobs() proc first, or to add a return ..() at the end of attack_self() proc. I chose the former.

**Comment by phil235** [comment]
the proc would stop only if you did `return ..()` , the fact that there's a return in the parent is irrelevant, it doesn't stop the child proc. 

(Also the return on line 125 is useless since procs return null by default)

File: code/game/objects/items/weapons/clown_items.dm [hunk 1]
```diff
@@ -143,13 +143,12 @@
 	item_state = "gold_horn"
 
 /obj/item/weapon/bikehorn/golden/attack()
-	..()
 	flip_mobs()
+	..()
```
Comment (ChangelingRain) [inline code/game/objects/items/weapons/clown_items.dm]: This needs to return ..() so attack works properly

**Comment by ChangelingRain** [comment]
The tl;dr is that the ..() sets spam_flag to 1 and the flip code won't run if spam_flag isn't FALSE in some form.

File: code/game/objects/items/weapons/clown_items.dm [hunk 2]
```diff
@@ -143,13 +143,12 @@
 	item_state = "gold_horn"
 
 /obj/item/weapon/bikehorn/golden/attack()
-	..()
 	flip_mobs()
+	return ..()
```
Comment (KorPhaeron) [inline code/game/objects/items/weapons/clown_items.dm]: Someone explain to me why this has to be return ..() instead of just ..() I am lost

Comment (optimumtact) [inline code/game/objects/items/weapons/clown_items.dm]: so the result of the parent call is returned, which helps keep the oo chain unbroken.

other options include . = ..()

Comment (KorPhaeron) [inline code/game/objects/items/weapons/clown_items.dm]: Did we ever decide if whether return or .=..() was better

Comment (phil235) [inline code/game/objects/items/weapons/clown_items.dm]: I think remie said . = ..() is faster.

**Pull request merged**